packages/server/src/handlers/BaseHandler.ts

```diff
@@ -101,4 +110,58 @@ export class BaseHandler extends EventEmitter {
 
     return {host: host as string, proto}
   }
+
+  protected getLocker(req: http.IncomingMessage) {
+    if (typeof this.options.locker === 'function') {
+      return this.options.locker(req)
+    }
+    return this.options.locker
+  }
+
+  protected async acquireLock(
+    req: http.IncomingMessage,
+    id: string,
+    context: CancellationContext
+  ) {
+    const locker = this.getLocker(req)
+    await locker?.lock(id, () => {
+      context.cancel()
+    })
+
+    return () => locker?.unlock(id)
```

Comment:
```suggestion
    if (locker == null) return
    await locker.lock(id, () => {
      context.cancel()
    })

    return () => locker.unlock(id)
```

What about this? Then we can get rid of these null checks.

Reply:
we still need a no-op function returned as the `unlock` otherwise it would break the logic on the handler when calling `unlock`.

I would rather sacrifice this method with null checks that cluttering the handlers with:

```
if (unlock) {
 try {
 
 ... logic
 } finally {
  await unlock()
 }
} else {
 ... logic
}
```

Reply:
I've done what you recommended as this would save a no-op call to the function when the locker is not provided.
On the handler side I've done the null check

`unlock?.()`